src/utils/salesEvidenceFilter.ts

```diff
@@ -0,0 +1,59 @@
+import { SalesEvidenceFilterParams, SaleTypeEnum, UnprocessedResultsFromCRM, MinMaxNumberType, MinMaxDateType } from '../types'
+
+function numberFilter (property: UnprocessedResultsFromCRM, filterType: string, filterValues: MinMaxNumberType) {
+    return typeof property[filterType] === 'number' && property[filterType] >= filterValues.min && property[filterType] <= filterValues.max
+}
+
+function formatDateToString (date: Date): string {
+    const dateYear = date.getFullYear()
+    const monthValuesStartAtOne = 1
+    let dateMonth: string | number = date.getMonth() + monthValuesStartAtOne
+    let dateDate: string | number = date.getDate()
+    const numberToLeftPadZero = 9
+    if (dateMonth <= numberToLeftPadZero) dateMonth = `0${dateMonth}`
+    if (dateDate <= numberToLeftPadZero) dateDate = `0${dateDate}`
+    return `${dateYear}-${dateMonth}-${dateDate}`
+}
+
+function dateFilter (property: UnprocessedResultsFromCRM, dateSold: MinMaxDateType): boolean {
+    if (typeof dateSold.min !== 'undefined' && typeof dateSold.max !== 'undefined') {
+        const minDate = formatDateToString(dateSold.min)
```

Comment:
conversion to string is not necessary. Date objects respect > and <
https://www.w3resource.com/javascript-exercises/javascript-date-exercise-5.php